apps/server/src/modules/task/service/submission.service.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { MikroORM } from '@mikro-orm/core';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Counted, Submission } from '@shared/domain';
+import { SubmissionRepo } from '@shared/repo';
+import { setupEntities, submissionFactory, taskFactory } from '@shared/testing';
+import { SubmissionService } from './submission.service';
+
+describe('Submission Service', () => {
+	let module: TestingModule;
+	let service: SubmissionService;
+	let submissionRepo: DeepMocked<SubmissionRepo>;
+	let orm: MikroORM;
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+
+		module = await Test.createTestingModule({
+			imports: [],
+			providers: [
+				SubmissionService,
+				{
+					provide: SubmissionRepo,
+					useValue: createMock<SubmissionRepo>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(SubmissionService);
+		submissionRepo = module.get(SubmissionRepo);
+	});
+
+	afterAll(async () => {
+		await orm.close();
+		await module.close();
+	});
+
+	beforeEach(() => {
+		jest.resetAllMocks();
+	});
+
+	it('should be defined', () => {
+		expect(service).toBeDefined();
+	});
+
+	describe('findAllByTask is called', () => {
+		const createParams = () => {
+			const task = taskFactory.buildWithId();
+
+			const submission1 = submissionFactory.buildWithId();
+			const submission2 = submissionFactory.buildWithId();
+			const submissions = [submission1, submission2];
+			const countedSubmissions: Counted<Submission[]> = [submissions, 2];
+
+			return { task, countedSubmissions };
+		};
+
+		describe('WHEN repo returns successfully', () => {
+			const setup = () => {
+				const { task, countedSubmissions } = createParams();
+
+				submissionRepo.findAllByTaskIds.mockResolvedValueOnce(countedSubmissions);
+
+				return { taskId: task.id, countedSubmissions };
+			};
+
+			it('should call findAllByTaskIds', async () => {
```

Comment:
I think this test is unnecessary because it tests inner workings and the repo call is not a side effect but directly effects the result. (Picking up the discussion we had a few weeks ago.)

Reply:
I guess a service section needs to be added, but the use case section is important for us here. So please see: https://hpi-schul-cloud.github.io/schulcloud-server/additional-documentation/nestjs-application/testing.html 

Please take further discussion to the team or arch chapter.